frontend/src/app/modules/imports/components/import_process/upload-file-step/upload-file-step.component.ts

```diff
@@ -42,19 +42,30 @@ export class UploadFileStepComponent implements OnInit {
     this.uploadForm = this.fb.group({
       file: [null, Validators.required],
       fileName: [null, [Validators.required, Validators.maxLength(this.maxFileNameLength)]],
+      dataset: [null, Validators.required],
     });
   }
 
   ngOnInit() {
+    this.setupDatasetSelect();
     this.step = this.route.snapshot.data.step;
     this.importData = this.importProcessService.getImportData();
     if (this.importData === null) {
-      this.datasetId = this.route.snapshot.queryParams['datasetId'];
+      this.uploadForm.patchValue({ dataset: this.route.snapshot.queryParams['datasetId'] });
     } else {
+      this.uploadForm.patchValue({ dataset: this.importData.id_dataset });
       this.fileName = this.importData.full_file_name;
     }
   }
 
+  setupDatasetSelect() {
```

Comment:
Consider adding type annotations to the methods and variables for better readability and maintainability of the code. [medium]